TradingBot/Services/Strategy.cs

```diff
@@ -1,33 +1,188 @@
+    {
+        var today = DateOnly.FromDateTime(_clock.UtcNow.UtcDateTime);
+        var result = await _marketDataSource.GetDataForSingleSymbolAsync(symbol, today, today, token);
+
+        return (result == null) ? null : result.First();
+    }
+
+    private static bool IsPriceDecreasing(IReadOnlyList<decimal> closingPrices, int maxDaysDecreasing)
+    {
+        var isPriceDecreasing = true;
+        maxDaysDecreasing = maxDaysDecreasing > closingPrices.Count - 1 ? closingPrices.Count - 1 : maxDaysDecreasing;
+
+        for (int i = 1; i <= maxDaysDecreasing; i++)
+        {
+            if (closingPrices[i] > closingPrices[0])
+            {
+                isPriceDecreasing = false;
+                break;
+            }
+        }
+
+        return isPriceDecreasing;
     }
+
+    private static bool IsPriceIncreasing(IReadOnlyList<decimal> closingPrices, int maxDaysDecreasing)
+    {
+        var isPriceIncreasing = true;
+        maxDaysDecreasing = maxDaysDecreasing > closingPrices.Count - 1 ? closingPrices.Count - 1 : maxDaysDecreasing;
+
+        for (int i = 1; i <= maxDaysDecreasing; i++)
+        {
+            if (closingPrices[i] < closingPrices[i - 1])
+            {
+                isPriceIncreasing = false;
+                break;
+            }
+        }
+
+        return isPriceIncreasing;
+    }
+
+    private static decimal CalculateAverageGrowthRate(IReadOnlyList<decimal> closingPrices)
+    {
+        decimal totalGrowth = 0;
+
+        for (int i = 1; i < closingPrices.Count; i++)
+        {
+            decimal dailyGrowth = ((closingPrices[i] - closingPrices[i - 1]) / closingPrices[i - 1]) * 100;
+            totalGrowth += dailyGrowth;
+        }
+
+        decimal averageGrowth = totalGrowth / (closingPrices.Count - 1);
+
+        return averageGrowth;
+    }
+
+    private List<TradingAction> GetBuyActions(IReadOnlyList<averageGrowthRate> growthRates, decimal cashAvaliable, int maxBuyCount, decimal topGrowingSymbolsBuyRatio)
+    {
+        var buyActions = new List<TradingAction>();
+
+        var topGrowingSymbols = growthRates
+            .OrderByDescending(x => x.AverageGrowthRate)
+            .Take(maxBuyCount)
+            .Select(x => (x.Symbol, x.Price))
+            .ToList();
+
+        for (int i = 0; i < topGrowingSymbols.Count; i++)
+        {
+            var symbol = topGrowingSymbols[i].Symbol;
+            var price = topGrowingSymbols[i].Price;
+            if(price >= 1)
+                price = Math.Round(price, 2);
+            else
+                price = Math.Round(price, 4);
+
+            var quantity = (int)(cashAvaliable * topGrowingSymbolsBuyRatio / price);
+
+            if (i == topGrowingSymbols.Count - 1)
+            {
+                quantity = (int)Math.Floor(cashAvaliable / price);
+            }
+
+            if (quantity * price > cashAvaliable || quantity <= 0)
```

Comment:
I think the first part here should be impossible? If that's the case and we still want to make sure, we should `throw` something with appropriate message if that happens.